### PR TITLE
Fix the sdk version and do not name a `ModelType` variable "Type"

### PR DESCRIPTION
**Pull request by dougbu**

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-    "projects": ["src", "test/WebSites", "samples"]
+    "projects": ["src", "test/WebSites", "samples"],
+    "sdk": {
+      "version": "1.0.0-preview2-003154"
```
Comment (dougbu) [inline global.json]: This is the SDK version the build downloads. Is it the correct one for the 1.0.3 milestone?

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs [hunk 2]
```diff
@@ -57,6 +60,18 @@ public static IEnumerable<object[]> ExpressionAndTexts
                         string.Empty
                     },
                     {
+                        (Expression<Func<TestModel, int>>)(model => models[0].SelectedCategory.CategoryId),
+                        "models[0].SelectedCategory.CategoryId"
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs]: `"s[0].SelectedCategory.CategoryId"` before fix

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs [hunk 3]
```diff
@@ -57,6 +60,18 @@ public static IEnumerable<object[]> ExpressionAndTexts
...
+                    },
+                    {
+                        (Expression<Func<TestModel, string>>)(model => modelTest.Name),
+                        "modelTest.Name"
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs]: `"Test.Name"` before fix

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs [hunk 4]
```diff
@@ -57,6 +60,18 @@ public static IEnumerable<object[]> ExpressionAndTexts
...
+                    },
+                    {
+                        (Expression<Func<TestModel, Type>>)(model => modelType),
+                        "modelType"
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs]: `"Type"` before fix

File: global.json (hunk 1 above)
Comment (pranavkm) [inline global.json]: It should be good for now. @JunTaoLuo, we'll need to update the CLI to the version that's shipping in the release (if any). Could you file a tracking item for this?

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs [hunk 5]
```diff
@@ -57,6 +60,18 @@ public static IEnumerable<object[]> ExpressionAndTexts
...
+                    },
+                    {
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs]: What happens if the expression uses something from a type or namespace called "Model" e.g `m => MyApp.Model.Constants.CopyrightYear` or some weird variant like this.

Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs]: We'd have to test those cases. But, this goes beyond the reported problem. So we should focus the follow-up discussion on `dev`.

**Comment by dougbu** [comment]
@pranavkm this doesn't show as signed off. Do you approve?

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs (hunk 5 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs]: Sounds good.

**Comment by pranavkm** [comment]
:shipit:

**Comment by dougbu** [comment]
- fa710e6 in `rel/1.0.3`
- cherry-picked test changes as a5c7a7e into dev
- did not merge `rel/1.0.3` branch into `dev` because this product code change does not apply and branches have diverged significantly

**Pull request merged**